modules/core/shared/src/test/scala/retry/PackageObjectSpec.scala

```diff
@@ -398,7 +398,7 @@ class PackageObjectSpec extends AnyFlatSpec {
 
     val finalResult = retryingOnFailuresAndAllErrors[String](
       policy,
-      _ == "will never happen",
+      s => Right(s == "will never happen"),
       onError,
       onError
     ) {
```

Comment:
I think it's worth adding a couple of tests to show what happens if `wasSuccessful` or `isWorthRetrying` raises an error.

Reply:
Done.